core/src/main/java/io/ultrabrew/metrics/data/BasicHistogramAggregator.java

```diff
@@ -0,0 +1,204 @@
+package io.ultrabrew.metrics.data;
+
+import io.ultrabrew.metrics.util.DistributionBucket;
+import java.util.Arrays;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import sun.misc.Unsafe;
+
+public class BasicHistogramAggregator extends ConcurrentIntTable implements Aggregator {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(BasicHistogramAggregator.class);
+  private static final String[] AGG_FIELDS = {"count", "sum", "min", "max"};
+  private static final int[] IDENTITY = {0, 0, Integer.MAX_VALUE, Integer.MIN_VALUE};
```

Comment:
array type needs to be `long[]` because these are statistics (the gauge part of the histogram vs counts within a bucket). since time is tracked in nanos, approximately 4 seconds will reach `Integer.MAX_VALUE` in nanos.